core/optaplanner-core-impl/src/main/java/org/optaplanner/core/api/score/IBendableScore.java

```diff
@@ -0,0 +1,70 @@
+package org.optaplanner.core.api.score;
+
+import java.io.Serializable;
+
+import org.optaplanner.core.api.score.buildin.bendable.BendableScore;
+
+/**
+ * Bendable score is a {@link Score} whose {@link #hardLevelsSize()} and {@link #softLevelsSize()}
+ * are only known at runtime.
+ * <p>
+ * Implementations must be immutable.
+ *
+ * @param <Score_> the actual score type to allow addition, subtraction and other arithmetic
+ * @see BendableScore
+ */
+public interface IBendableScore<Score_ extends IBendableScore<Score_>>
```

Comment:
This bugs me. This would have been ideally called `BendableScore`, but the impl of this interface is already called `BendableScore`.
Any other ideas?